apps/api/src/app/environments-v1/usecases/output-renderer-usecases.ts

```diff
@@ -0,0 +1,143 @@
+// Base interface for all renderers
+import { render } from '@maily-to/render';
+import {
+  ChatRenderOutput,
+  EmailRenderOutput,
+  InAppRenderOutput,
+  PushRenderOutput,
+  RedirectTargetEnum,
+  SmsRenderOutput,
+  TipTapNode,
+} from '@novu/shared';
+import { z } from 'zod';
+import { ExpendEmailEditorSchemaUseCase } from '../../environments/render/email-schema-extender';
+
+class RenderCommand {
+  controlValues: Record<string, unknown>;
+}
+
+export class EmailOutputRendererUseCase {
+  async execute(renderCommand: RenderCommand): Promise<EmailRenderOutput> {
+    const parse = EmailStepControlSchema.parse(renderCommand.controlValues);
+    const schema = parse.emailEditor as TipTapNode;
+    const expandedSchema = new ExpendEmailEditorSchemaUseCase().execute({ schema });
```

Comment:
As above, this should be injected via the constructor via an `@Injectable` decorator and supplied to the consuming NestJS module.